pints/plot.py

```diff
@@ -8,6 +8,201 @@
 #
 from __future__ import division
 
+def log_likelihood_1d(log_likelihood, parameters, boundaries=None, n_eva=20, chain=None):
+    """
+    Takes a set of parameters as input and creates and returns a plot showing 
+    the log-likelihood function of each parameter with all the other 
+    parameters held at the given set of parameters.
+    
+    Arguments:
+
+    `log_likelihood`
+        A class (or that extends from) :class:`pints.LogLikelihood` of the 
+        problem of interest.
+    `parameters`
+        A set of parameters, which the `log_likelihood` will be evaluated at, 
+        with the same dimension as the `log_likelihood` will accept.
+    `boundaries`
+        A class (or that extends from) :class:`pints.Boundaries` which defines 
+        the boundaries of each parameters.  
+    `n_eva`
+        (Optional) The number of evaluation of each parameter.
+    `chain`
+        (Optional) A markov chain of shape `(samples, dimension)`, where 
+        `samples` is the number of samples in the chain and `dimension` is the 
+        number of parameters. This will be used to define the boundaries. Only 
+        provide either `chain` or `boundaries`, but not both.
+
+    Returns a `matplotlib` figure object and axes handle.
+    """
+    import pints
+    import numpy as np
+    try:
+        import matplotlib.pyplot as plt
+    except:
+        raise ImportError("The method pints.plot.trace requires matplotlib")
```

Comment:
Are we doing this in the other methods? It seems a bit redundant, since you're catching an ImportError saying matplotlib can't be imported, and then raising a new ImportError to say matplotlib can't be imported

Reply:
Can you split it into two checks (one check for both, one check for neither) and give the appropriate error message? Will make it a bit easier for people to debug :-)